exoscale/api/compute.py

```diff
@@ -1165,14 +1215,12 @@ def instances(self):
             latency.
         """
 
-        try:
-            [res] = self.compute.cs.getInstancePool(
-                id=self.id, zoneid=self.zone.id, fetch_list=True
-            )
-            for i in res.get("virtualmachines", []):
-                yield Instance._from_cs(self.compute, i, zone=self.zone)
-        except CloudStackApiException as e:
-            raise APIException(e.error["errortext"], e.error)
+        res = self.compute._v2_request(
+            "GET", "/instance-pool/" + self.id, self.zone.name
+        )
+
+        for i in res["instances"]:
+            yield self.compute.get_instance(self.zone, id=i["id"])
```

Comment:
You will perform an API call for every instance to get full details
From public-api PoV it is not what `list-instances` is supposed to do.
Why changing the behavior in the Python client ?

Reply:
You're right, but this property is not equivalent to `list_instances()`: it returns the list of `Instance` objects referenced by the Instance Pool, for which in order to have the "complete" version of the resources must be retrieved using the `get-instance` API operation because the `list-instances` operation only returns shallow resources.